f5_openstack_agent/lbaasv2/drivers/bigip/resource_helper.py

```diff
@@ -246,3 +246,26 @@ def _collection(self, bigip):
                       "resource %s", self.resource_type)
             raise KeyError("No collection available for %s" %
                            (self.resource_type))
+
+    def get_stats(self, bigip,  name=None, partition=None, stats=None):
```

Comment:
stats must default to [] not None.  Line 266 will fail with the following:

>>> stats = None
>>> for stat in stats:
...   print stat
...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'NoneType' object is not iterable